### ISSUE TITLE
[BUG] ValueErrors on MacBook on >=v1.6.22

### ISSUE THREAD

**Issue by mgovers** [PowerGridModel/power-grid-model#466]
**Describe the bug**

The data types on MacBook changed in `v1.6.22`. As a result, `pandas.DataFrame` representations of the PGM output data may not be correct and/or consistent.

Further investigation will be done in pair programming session.

**Input Data Validity**

(private data)

**Data Dump**

(private data)

**To Reproduce**

```py
# (private data)
df = pd.merge(output_data, ...)
df.reindex(...)  # <- raises ValueError: cannot reindex on an axis with duplicate labels
```

**Expected behavior**

(private data)

**Additional context**

It affects MacBooks with M1 processors. It seems to be closely related to https://github.com/PowerGridModel/power-grid-model/compare/v1.6.21..v1.6.22

**Comment by TonyXiang8787** [comment]
@mgovers there is no description of the bug?

**Comment by mgovers** [comment]
Will be added to in the near future. I asked the reporter to add to this issue

Op 11 jan. 2024 10:20 schreef Tony Xiang ***@***.***>:

@mgovers<https://github.com/mgovers> there is no description of the bug?

**Comment by mgovers** [comment]
unit tests + validation tests pass on Jean's MacBook. Working on a minimal repro case

**Comment by mgovers** [comment]
Both independent reporters @Thijss and Jean confirmed the issue to be resolved by #477 (released in `v1.6.55`).
